### PR TITLE
Reuse PrivateKey, PublicKey and Signature from eth_keys

### PR DESCRIPTION
**Pull request by yglukhov**

### CODE REVIEW

File: tests/testecc.nim [hunk 1]
```diff
@@ -53,7 +54,7 @@ suite "ECC/ECDSA/ECDHE tests suite":
     var expectm = """
       8ac7e464348b85d9fdfc0a81f2fdc0bbbb8ee5fb3840de6ed60ad9372e718977"""
     var secret: array[KeyLength, byte]
-    var s = keccak256.digest("ecdhAgree").data
+    var s = initPrivateKey(keccak256.digest("ecdhAgree").data)
```
Comment (cheatfate) [inline tests/testecc.nim]: `initPrivateKey()` is a very bad name for hex conversion function, for me `initPrivateKey()` its create new private key.
First of all we will not store private keys as hexadecimal strings, and there is already present `getPrivateKey` which actually do same thing.

Comment (yglukhov) [reply tests/testecc.nim]: It is not hex string here, `keccak256.digest("ecdhAgree").data ` is `array[32, byte]`.

**Comment by yglukhov** [comment]
@cheatfate, rebased.

**Comment by cheatfate** [comment]
After some investigation of `eth_keys` code, i wish not to merge this PR, until `eth_keys` pass review.

**Comment by yglukhov** [comment]
@cheatfate, those are only 2 functions in `eth_keys` that are relevant to this PR: `initPrivateKey(data: array[32, byte])` and `public_key(p: PrivateKey)`.

**Comment by cheatfate** [comment]
@yglukhov problem is not in your code, your code is good enough, problem is in the `eth_keys`. You trying to use functions from `eth_keys` but this is not safe and making your code not safe. `eth_keys` using exceptions everywhere, but we discussed this already.

**Comment by cheatfate** [comment]
@yglukhov as i said i will not accept this PR, until `nim-eth-keys` will not pass review.
Using `initPrivateKey` is not safe because there no checks performed on incoming data.

**Comment by yglukhov** [comment]
@cheatfate 
what exactly do you mean by "pass review"? It has passed my review enough to get this pr merged.

`initPrivateKey` is used in this pr in places where there was no validation of incoming data. It is like 100% equivalent, I bet even the assembly is the same.

**Comment by cheatfate** [comment]
eth_keys did not passed any reviews, in some circumstances it will generate unrecoverable crashes (generated by libsecp256k1).

**Comment by cheatfate** [comment]
According to current situation (recent talks to deprecate `ttmath`) i think it would be better to move `ecc.nim` as the only backend to `eth_keys`, because it more safe.

**Comment by yglukhov** [comment]
Closing this, since `eth_keys` was refactored.

**Pull request closed without merging**